bundles/org.openhab.binding.ntp/src/main/java/org/openhab/binding/ntp/internal/handler/NtpHandler.java

```diff
@@ -54,217 +54,183 @@
  * to one of the channels.
  *
  * @author Marcel Verpaalen - Initial contribution OH2 ntp binding
- * @author Thomas.Eichstaedt-Engelen OH1 ntp binding (getTime routine)
+ * @author Thomas.Eichstaedt - Engelen OH1 ntp binding (getTime routine)
  * @author Markus Rathgeb - Add locale provider
  * @author Erdoan Hadzhiyusein - Adapted the class to work with the new DateTimeType
+ * @author Laurent Garnier - null annotations, TimeZoneProvider, configuration settings cleanup
  */
-
+@NonNullByDefault
 public class NtpHandler extends BaseThingHandler {
 
-    private final Logger logger = LoggerFactory.getLogger(NtpHandler.class);
-
     /** timeout for requests to the NTP server */
     private static final int NTP_TIMEOUT = 30000;
 
     public static final String DATE_PATTERN_WITH_TZ = "yyyy-MM-dd HH:mm:ss z";
+    private static final DateTimeFormatter DATE_FORMATTER_WITH_TZ = DateTimeFormatter.ofPattern(DATE_PATTERN_WITH_TZ);
 
-    /** for logging purposes */
-    private final DateFormat SDF = new SimpleDateFormat(DATE_PATTERN_WITH_TZ);
+    private final Logger logger = LoggerFactory.getLogger(NtpHandler.class);
 
     /** for publish purposes */
     private DateTimeFormatter dateTimeFormat = DateTimeFormatter.ofPattern(DATE_PATTERN_WITH_TZ);
 
-    private final LocaleProvider localeProvider;
+    private @Nullable NtpThingConfiguration configuration;
```

Comment:
Since all values have reasonable defaults, you can skip null checks by using 
```suggestion
    private NtpThingConfiguration configuration = new NtpThingConfiguration();
```

Reply:
Ok, done.